src/mds/Locker.cc

```diff
@@ -3634,9 +3650,19 @@ void Locker::_do_snap_update(CInode *in, snapid_t snap, int dirty, snapid_t foll
   if (xattrs) {
     dout(7) << " xattrs v" << i->xattr_version << " -> " << m->head.xattr_version
 	    << " len " << m->xattrbl.length() << dendl;
-    i->xattr_version = m->head.xattr_version;
-    auto p = m->xattrbl.cbegin();
-    decode(*px, p);
+    __u32 total = 0;
+    for (const auto &p : *px)
+      total += (p.first.length() + p.second.length());
+    if ((total + m->xattrbl.length()) >= g_conf()->mds_max_xattr_pairs_size) {
+      dout(10) << "xattr kv pairs size too big. current size: " << total
+             << " new size: " << (total + m->xattrbl.length()) << dendl;
+      // Ignore new xattr (!!!) but increase xattr version
+      i->xattr_version = m->head.xattr_version + 1;
```

Comment:
Not very sure we should increase the version here, since nothing changed.

Reply:
The reason I had this was that we would be silently dropping changes done by a client, which would have buffered these xattrs with a specific version X.  If another client would set an xattr and the version would be set to X, the first client could get confused.  But since we're not allowing `Xx` anymore on these old clients, I guess this line can be dropped.